### PR TITLE
Basic version of showing threaded messages in a stack

### PR DESCRIPTION
**Pull request by MartinJibber**
IMG_0564

### CODE REVIEW

File: iOS/Flows/Home/_New/ConversationViewControllerDataSource+CellRegistration.swift [hunk 1]
```diff
@@ -25,16 +25,18 @@ extension ConversationCollectionViewDataSource {
...
-            // Configure the cell
-            cell.contentView.set(backgroundColor: .red)
 
             if message.type == .deleted {
-                cell.textView.text = "DELETED"
-                cell.replyCountLabel.text = nil
+                cell.setIsDeleted()
             } else {
-                cell.textView.text = message.text
-                cell.replyCountLabel.setText("\(message.latestReplies.count)")
```
Comment (MartinJibber) [inline iOS/Flows/Home/_New/ConversationViewControllerDataSource+CellRegistration.swift:36]: Moving some of the cell setup code into the cell itself. Importantly, the cell still does not know anything about chat message objects or the data model in general.

File: iOS/Flows/Home/_New/new_ConversationViewController.swift [hunk 2]
```diff
@@ -198,12 +198,7 @@ extension new_ConversationViewController: SwipeableInputAccessoryViewDelegate {
 
     private func send(object: Sendable) async {
         do {
-            guard let latestMessage = self.conversationController.messages.last else { return }
-
-            #warning("Revert")
-            try await self.conversationController.createNewReply(for: latestMessage.id,
-                                                                    with: object)
-//            try await self.conversationController?.createNewMessage(with: object)
```
Comment (MartinJibber) [inline iOS/Flows/Home/_New/new_ConversationViewController.swift:206]: This was debug code to send threaded messages.

File: iOS/Flows/Home/_New/new_MessageCell.swift [hunk 3]
```diff
@@ -10,16 +10,30 @@ import Foundation
...
 class new_MessageCell: UICollectionViewCell {
 
-    let textView = TextView()
-    let replyCountLabel = Label(font: .regular)
+    private let messageContainerView = UIView()
+    private let messageTextView = TextView()
+
+    private let repliesContainerView = UIView()
+    private let replyCountLabel = Label(font: .regular)
+    private let repliesTextView = TextView()
```
Comment (MartinJibber) [inline iOS/Flows/Home/_New/new_MessageCell.swift:18]: Created some rounded containers to hold the main message and the latest threaded message.

File: iOS/Flows/Home/_New/ConversationViewControllerDataSource+CellRegistration.swift (hunk 1 above)
Comment (Benjamindodgson) [reply iOS/Flows/Home/_New/ConversationViewControllerDataSource+CellRegistration.swift:36]: Lets be carefully to not store any state of the model on the cell itself. Im pretty sure it has a different lifecycle than the registration.

File: iOS/Flows/Home/_New/new_MessageCell.swift [hunk 4]
```diff
@@ -29,12 +43,54 @@ class new_MessageCell: UICollectionViewCell {
...
+        self.repliesTextView.centerOnXAndY()
+    }
+
+    func setMessage(_ text: String) {
+        self.messageTextView.text = text
+
+        self.setNeedsLayout()
+    }
+
+    func setReplies(_ replies: [String]) {
```
Comment (Benjamindodgson) [inline iOS/Flows/Home/_New/new_MessageCell.swift:74]: I would imagine this would eventually be Messagable array?

**Review by Benjamindodgson — APPROVED** [review]

File: iOS/Flows/Home/_New/new_MessageCell.swift (hunk 4 above)
Comment (MartinJibber) [reply iOS/Flows/Home/_New/new_MessageCell.swift:74]: Yeah that's a good idea.

**Pull request merged**